book.toml

```
@@ -180,8 +180,8 @@ use-boolean-and = true
 "ownership/moves-function-calls.html" = "../memory-management/move.html"
 "ownership/shared-unique-borrows.html" = "../borrowing/shared.html"
 "pattern-matching/destructuring-arrays.html" = "../tuples-and-arrays/destructuring.html"
-"pattern-matching/destructuring-enums.html" = "../pattern-matching/destructuring.html"
-"pattern-matching/destructuring-structs.html" = "../pattern-matching/destructuring.html"
+"pattern-matching/destructuring-enum.html" = "../pattern-matching/destructuring-enums.html"
+"pattern-matching/destructuring-struct.html" = "../pattern-matching/destructuring-structs.html"
```

Please see the comment above the `redirect` table in the TOML file.

In this case you're renaming

- from `src/pattern-matching/destructuring.md`
- to `src/pattern-matching/destructuring-enums.md`

This means that the

- old URL is `pattern-matching/destructuring.html`
- and the new URL is `pattern-matching/destructuring-enums.html`

The `redirect` table is a bit weird, but when you have `somewhere/old.html = elsewhere/new.html`, you instruct `mdbook build` to create a small `somewhere/old.html` file with some JavaScript which will redirect the visitor to `elsewhere/new.html`.

Because the redirect happens when you visit `somewhere/old.html`, it is important to make it relative to `somewhere/old.html`. If you redirect to `elsewhere/new.html`, the final URL becomes `somewhere/elsewhere/new.html`.

In this case, you should redirect like this:

```suggestion
"pattern-matching/destructuring.html" = "destructuring-enums.html"
```

You should be able to double-check this locally by noticing that `mdbook build` outputs the redirection files. I think it might also work in `mdbook serve`?